### PR TITLE
collection update route test and fixes

### PR DESCRIPTION
**Pull request by bennycio**
Changes:

- **Added** Collection Update Route test
- **Changed** collection update to accept an interface and added multiple update structs for updating specific parts of a collection, each corresponding to an endpoint
- **Changed** collection update endpoint to bind query not json

### CODE REVIEW

**Review by kaitoo1 — APPROVED** [review]

File: server/t__routes_collection_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	"github.com/mikeydub/go-gallery/persist"
+	"github.com/mikeydub/go-gallery/runtime"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUpdateCollectionById_Success(t *testing.T) {
+	assert := assert.New(t)
+
+	// seed DB with nft
```
Comment (kaitoo1) [inline server/t__routes_collection_test.go:19]:
```suggestion
	// seed DB with collection
```

Comment (kaitoo1) [reply server/t__routes_collection_test.go:19]: Similar changes for the rest of the comments in this test

File: server/t__routes_collection_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/mikeydub/go-gallery/persist"
+	"github.com/mikeydub/go-gallery/runtime"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUpdateCollectionById_Success(t *testing.T) {
```
Comment (kaitoo1) [inline server/t__routes_collection_test.go:16]: It looks like this test specifically tests the Update Name functionality, maybe this test can be renamed to `TestUpdateCollectionNameById_Success`

Comment (bennycio) [reply server/t__routes_collection_test.go:16]: For sure!

**Pull request merged**